vassal-app/src/main/java/VASSAL/build/module/PlayerRoster.java

```diff
@@ -622,13 +622,36 @@ protected boolean allSidesAllocated() {
 
   /**
    * @param side Name of a side to see if it's a "solo side"
-   * @return True if the side is "Solitaire", "Solo", "Moderator", or "Referee"
+   * @return True if the side is "Solo" or begins "Solitaire", "Solo:", "Moderator", "Referee" or a Side
+   * found in the Global Translatable Message Property: VassalSkipSideList (a comma-separated list).
+   * To test specifically for Referee/Moderator, use isRefereeSide()
    */
   public static boolean isSoloSide(String side) {
-    return Resources.getString("PlayerRoster.solitaire").equals(side) ||
-           Resources.getString("PlayerRoster.solo").equals(side) ||
-           Resources.getString("PlayerRoster.moderator").equals(side) ||
-           Resources.getString("PlayerRoster.referee").equals(side);
+    // FIXME: A future version of Vassal may provide a Side Type attribute in Definition of Sides that would allow this function to be rationalised.
+    // See also isRefereeSide()
+    return side.startsWith(Resources.getString("PlayerRoster.solitaire")) ||
+            side.equals(Resources.getString("PlayerRoster.solo")) ||
+            side.startsWith(Resources.getString("PlayerRoster.solo") + ":") ||
+            side.startsWith(Resources.getString("PlayerRoster.moderator")) ||
+            side.startsWith(Resources.getString("PlayerRoster.referee")) ||
+            side.startsWith(Resources.getString("PlayerRoster.auxiliarySidePrefix")) || // auxiliary side prefix
+            side.startsWith(SOLITAIRE) || // these checks needed in case a translated module has not localised these side names
+            side.startsWith(SOLO) ||
+            side.startsWith(SOLO + ":") ||
+            side.startsWith(MODERATOR) ||
+            side.startsWith(REFEREE);
+  }
+
+  /**
+   * @param side Name of a side to see if it's a "referee side"
+   * @return True if the side is begins "Moderator" or "Referee"
+   * To test for non-player sides in general, use isSoloSide()
+   */
+  public static boolean isRefereeSide(String side) {
+    return side.startsWith(Resources.getString("PlayerRoster.moderator")) ||
```

Comment:
Again I'm fuzzy on the change to startsWith and also the need to compare to the raw strings

Reply:
The raw strings comparison arises from a concern that I have about this along the following lines...

1. Say a module has partial translation but not in the Definition of Sides component. This is not so outlandish, as translated PlayerSides can open up other complications if the module depends on PlayerSide, so I've found.
2. The translation resources for solitaire etc exist regardless.... therefore when Vassal isSoloSide() scans the sides array it will attempt to find the translated side name. This will fail if the solo side names don't match in translation, but the raw string will then catch it.

Have I got that right?